lib/pbench/test/unit/agent/test_tool_data_sink.py

```diff
@@ -284,13 +288,194 @@ def mock_make_server(host, port, app, *args, **kwargs):
                 )
             with pytest.raises(AssertionError) as exc:
                 # Call it again to verify the assertion fires
-                self.wsgi_server.run(_test_app)
+                wsgi_server.run(_test_app)
             assert "'run' method called twice" in str(exc.value), f"{exc.value}"
             # No logs should have been emitted.
             assert len(caplog.records) == 3
 
-    def test_stop_and_wait(self, server, caplog):
+    def test_stop_and_wait(self, caplog):
         """test_stop_and_wait - verify the operation of run() in conjunction
-        with stop() and wait() methods.
+        with stop() and wait() methods from separate threads.
+
+        There are a number of scenarios for the order of operations between
+        threads that we need to test.  We list them here using "MainThr" as the
+        name of the "main thread" which _creates_ the WSGI thread, and "WsgiThr"
+        as the name of the created WSGI thread invoking the "run" method.
+
+        References:
+            .wait() called at
+                line 184 (in .stop() method)
+                line 833 (in __enter__ method)
+            .stop() called at
+                line 886 (in __exit__ method)
```

Comment:
I don't think including line numbers here is a good idea; they'll change, and get out of sync, and these notes will become more confusing than helpful. How many times does `stop()` or `__enter_()` call `wait()` anyway? If there are multiple calls, or if you really feel the reader needs a specific reference, calling out line comments like `# Unit test reference 1` might be a more stable way to accomplish that.

Reply:
Fixed.